framework/db/Command.php

```diff
@@ -829,30 +829,32 @@ public function execute()
     {
         $sql = $this->getSql();
 
-        $rawSql = $this->getRawSql();
-
-        Yii::info($rawSql, __METHOD__);
+        $log = $this->db->enableQueryLog;
+        if ($log) {
+            $rawSql = $this->getRawSql();
+            Yii::info($rawSql, __METHOD__);
+            $token = $rawSql;
+        }
 
         if ($sql == '') {
             return 0;
         }
 
         $this->prepare(false);
 
-        $token = $rawSql;
         try {
-            Yii::beginProfile($token, __METHOD__);
+            $log && Yii::beginProfile($token, __METHOD__);
```

Comment:
note that the main point of disabling is to avoid calling getRawSql() which may be heavy. separate methods would not do much except adding an if statement.